packages/remix-server-runtime/responses.ts

```diff
@@ -1,5 +1,13 @@
-export type JsonFunction = <Data>(
-  data: Data,
+type Json =
+  | null
+  | boolean
+  | number
+  | string
+  | Array<Json>
+  | { [key: string]: Json };
+
+export type JsonFunction = (
+  data: Json,
```

Comment:
```suggestion
export type JsonFunction = <Data extends Json>(
  data: Data,
```
This way, `json` will support a generic, but the generic needs to match the Json type. Without this, there's no way to pass a generic like right now and this doing that change will cause a breaking change.

Technically, doing this will also be a breaking change but since it's more similar to the correct behavior you could consider the previous one allowing things like Date a bug.

Reply:
Good callout 👍